Library/Homebrew/utils.rb

```diff
@@ -221,9 +221,8 @@ def quiet_system(cmd, *args)
 end
 
 def curl(*args)
-  brewed_curl = HOMEBREW_PREFIX/"opt/curl/bin/curl"
-  curl = if MacOS.version <= "10.6" && brewed_curl.exist?
-    brewed_curl
+  curl = if MacOS.version <= "10.6" && Formula["curl"].installed?
+    Formula["curl"].opt_bin/"curl"
```

Comment:
The previous way of doing this was intended to avoid instantiating formula objects (in particular because it could cause issues during the tests, where the `curl` formula is unavailable). Mind leaving it the old way?